.cloudbees/testing/action.yml

```diff
@@ -45,9 +45,34 @@ inputs:
   dry-run:
     description: 'Simulate an upgrade, without actually doing it'
     default: 'false'
+  registry-configuration:
+    description: >
+      CloudBees registry configuration file containing the registries to use for loading images.
+      By default it uses the file containing the registries configured under 'Integrations' in the CloudBees platform.
+    default: ${{ cloudbees.registries }}
 runs:
   using: composite
   steps:
+    - id: mirror-chart-location
+      name: Fetch chart location
+      if: ${{ inputs.registry-configuration != '' }}
+      uses: docker://alpine:3.20
+      run: |
+        set -euo pipefail
+        if echo "${{ inputs.chart-location }}" | grep -q '^oci://'; then
+          echo -n "${{ inputs.chart-location }}" | sed 's|^oci://||' > ${CLOUDBEES_OUTPUTS}/chart-mirrors-location
+        else
+          echo -n 'x' > ${CLOUDBEES_OUTPUTS}/chart-mirrors-location
+        fi
+
+    - id: remap
+      name: Fetch registry mirrors
+      if: ${{ steps.mirror-chart-location.outputs.chart-mirrors-location != 'x' }}
+      uses: docker://public.ecr.aws/l7o7z1g8/services/registry-config:0.0.22
+      with:
+        entrypoint: /registry-config
+        args: --config "${{ inputs.registry-configuration }}" resolve "${{ steps.mirror-chart-location.outputs.chart-mirrors-location }}" resolved_registries.out
+
     - id: helm-install
       name: Install a Helm chart
       uses: docker://020229604682.dkr.ecr.us-east-1.amazonaws.com/actions/helm-install-action:${{ action.scm.sha }}
```

Comment:
It would be disadvantageous bloating pipeline that much by adding two more steps for each helm install Action call (and as we know there may be multiple of them within a job.
Thus, to reduce the number of steps, can you not move all of that logic into the existing step that uses a custom image and therefore copy the `registry-config` binary into the `helm-install-action` image?

Reply:
Done - updated.